lib/ibm_cloud_sdk_core/icp4d_token_manager.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+require("http")
+require("json")
+require("rbconfig")
+require_relative("./version.rb")
+require_relative("./jwt_token_manager")
+
+module IBMCloudSdkCore
+  # Class to manage ICP4D Token Authentication
+  class ICP4DTokenManager < JWTTokenManager
+    def initialize(icp4d_url: nil, username: nil, password: nil, access_token: nil)
+      icp4d_url += "/v1/preauth/validateAuth"
+      @username = username
+      @password = password
+      super(url: icp4d_url, user_access_token: access_token)
```

Comment:
I think you had a check in here before to throw an error if URL is not passed in (and access token is not passed in)?

I know the same check exists in the base service but since this is its own module, I think it should have a check also.

Reply:
yeah, I moved it because we are checking for it outside. I can add it bach here too.